src/datachain/lib/dc.py

```diff
@@ -1624,6 +1625,208 @@ def subtract(  # type: ignore[override]
             )
         return self._evolve(query=self._query.subtract(other._query, signals))  # type: ignore[arg-type]
 
+    def diff(  # noqa: PLR0912, C901
+        self,
+        other: "DataChain",
+        on: Union[str, Sequence[str]],
+        right_on: Optional[Union[str, Sequence[str]]] = None,
+        compare: Optional[Union[str, Sequence[str]]] = None,
+        right_compare: Optional[Union[str, Sequence[str]]] = None,
+        added: bool = True,
+        deleted: bool = True,
+        modified: bool = True,
+        unchanged: bool = False,
+        status_col: Optional[str] = None,
+    ) -> "Self":
```

Comment:
It would be great if you can extract these 100 lines of logic outside of dc.py while keep the method with the docs here 🙂

Reply:
Extracted to `datachain.lib.diff.py`